dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/reservedvalue/hibernate/HibernateReservedValueStore.java

```diff
@@ -0,0 +1,170 @@
+        List<String> values )
+    {
+        BatchHandler<ReservedValue> batchHandler = batchHandlerFactory
+            .createBatchHandler( ReservedValueBatchHandler.class ).init();
+
+        List<String> availableValues = getIfAvailable( reservedValue, values );
+
+        List<ReservedValue> toAdd = new ArrayList<>();
+
+        availableValues.forEach( ( value ) -> {
+            ReservedValue rv = new ReservedValue(
+                reservedValue.getOwnerObject(),
+                reservedValue.getOwnerUid(),
+                reservedValue.getKey(),
+                value,
+                reservedValue.getExpires()
+            );
+
+            batchHandler.addObject( rv );
+            toAdd.add( rv );
+
+        } );
+
+        batchHandler.flush();
+
+        return toAdd;
+    }
+
+    @Override
+    public List<ReservedValue> getIfReservedValues( ReservedValue reservedValue,
+        List<String> values )
+    {
+        return (List<ReservedValue>) getCriteria()
+            .add( Restrictions.eq( "ownerObject", reservedValue.getOwnerObject() ) )
+            .add( Restrictions.eq( "ownerUid", reservedValue.getOwnerUid() ) )
+            .add( Restrictions.eq( "key", reservedValue.getKey() ) )
+            .add( Restrictions.in( "value", values ) )
+            .list();
+    }
+
+    @Override
+    public int getNumberOfUsedValues( ReservedValue reservedValue )
+    {
+        Long count = (long) getQuery( "SELECT count(*) FROM ReservedValue WHERE owneruid = :uid AND key = :key" )
+            .setParameter( "uid", reservedValue.getOwnerUid() )
+            .setParameter( "key", reservedValue.getKey() )
+            .getSingleResult();
+
+        if ( Objects.valueOf( reservedValue.getOwnerObject() ).equals( TRACKEDENTITYATTRIBUTE ) )
+        {
+            count += (long) getQuery(
+                "SELECT count(*) " +
+                    "FROM TrackedEntityAttributeValue " +
+                    "WHERE attribute = " +
+                    "( FROM TrackedEntityAttribute " +
+                    "WHERE uid = :uid ) " +
+                    "AND value LIKE :value " )
+                .setParameter( "uid", reservedValue.getOwnerUid() )
+                .setParameter( "value", reservedValue.getValue() )
+                .getSingleResult();
+        }
+
+        return count.intValue();
+    }
+
+    @Override
+    public void removeExpiredReservations()
+    {
+        getQuery( "DELETE FROM ReservedValue WHERE expires < :now" )
+            .setParameter( "now", new Date() )
+            .executeUpdate();
+    }
+
+    @Override
+    public boolean useReservedValue( String ownerUID, String value )
+    {
+        return getQuery( "DELETE FROM ReservedValue WHERE owneruid = :uid AND value = :value" )
+            .setParameter( "uid", ownerUID )
+            .setParameter( "value", value )
+            .executeUpdate() == 1;
+    }
+
+    // Helper methods:
```

Comment:
Adjust style of "Helper methods". Not consistent with others. Maybe we should have a style convention for this?